### PR TITLE
Move ENPROTECTED to Interop Richedit

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Move ENPROTECTED to Interop Richedit.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6399)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs [hunk 1]
```diff
@@ -3443,22 +3443,11 @@ internal unsafe void WmReflectNotify(ref Message m)
 
                     case EN.PROTECTED:
                         {
-                            NativeMethods.ENPROTECTED enprotected;
-
-                            //On 64-bit, we do some custom marshalling to get this to work. The richedit control
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/RichTextBox.cs:3448]: Why is this no longer a problem?

**Review by Tanya-Solyanik — COMMENTED** [review]
Please see why the build is failing

File: src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Richedit/ENPROTECTEDTests.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+using Xunit;
+using static Interop.Richedit;
+
+namespace System.Windows.Forms.Primitives.Tests.Interop.Richedit
+{
+    public class ENPROTECTEDTests
+    {
+        [Fact]
+        public unsafe void Enprotected_Size_Get_ReturnsExpected()
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/Richedit/ENPROTECTEDTests.cs:13]: Please break it out as two independent tests, like we do for 
https://github.com/dotnet/winforms/blob/64c7a3b78d623117c330ea778a34f436da83e538/src/System.Windows.Forms.Primitives/tests/UnitTests/Interop/ComCtl32/TASKDIALOGCONFIGTests.cs#L10
Also please add a test for the struct layouts, as sometimes there could be subtle differences on each platform.

File: src/System.Windows.Forms.Primitives/src/Interop/Richedit/Interop.ENPROTECTED.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+    internal static partial class Richedit
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        public struct ENPROTECTED
+        {
+            public User32.NMHDR nmhdr;
+            public uint msg;
+            public UIntPtr wParam;
+            public IntPtr lParam;
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/Richedit/Interop.ENPROTECTED.cs:17]:
```cpp
typedef struct _enprotected
{
	NMHDR nmhdr;
	UINT msg;
	WPARAM wParam;
	LPARAM lParam;
	CHARRANGE chrg;
} ENPROTECTED;
```
I _think_ these could be
```suggestion
            public nuint msg;
            public nuint wParam;
            public nint lParam;
```



@JeremyKuhne thoughts?

**Comment by Tanya-Solyanik** [comment]
@gpetrou  - this PR is getting out of date, most changes are being done in https://github.com/dotnet/winforms/pull/6962 which is very close to get merged. Please review that PR and either close or updated this one.

**Comment by RussKie** [comment]
@gpetrou thank you for your work, but I'm closing this in favour of https://github.com/dotnet/winforms/pull/6962.

**Pull request closed without merging**